library/src-bootstrapped/scala/Tuple.scala

```diff
@@ -428,53 +336,15 @@ sealed trait NonEmptyTuple extends Tuple {
       }
     }
 
-  inline def stagedApply(n: Int): Elem[this.type, n.type] =
-    ${ StagedTuple.applyStaged[this.type, n.type](
-       'this, constValueOpt[Size[this.type]],
-       'n, constValueOpt[n.type]) }
-}
-
-object NonEmptyTuple {
-  import Tuple._
-
-  def dynamicHead[This <: NonEmptyTuple] (self: This): Head[This] = {
-    type Result = Head[This]
-    val res = (self: Any) match {
-      case self: Tuple1[_] => self._1
-      case self: Tuple2[_, _] => self._1
-      case self: Tuple3[_, _, _] => self._1
-      case self: Tuple4[_, _, _, _] => self._1
-      case self: TupleXXL => self.elems(0)
-      case self: Product => self.productElement(0)
-    }
-    res.asInstanceOf[Result]
-  }
-
-  def dynamicTail[This <: NonEmptyTuple] (self: This): Tail[This] = {
-    type Result = Tail[This]
-    val res = (self: Any) match {
-      case self: Tuple1[_] => ()
-      case self: Tuple2[_, _] => Tuple1(self._2)
-      case self: Tuple3[_, _, _] => Tuple2(self._2, self._3)
-      case self: Tuple4[_, _, _, _] => Tuple3(self._2, self._3, self._4)
-      case _ => dynamicFromArray[Result](dynamicToArray(self).tail)
-    }
-    res.asInstanceOf[Result]
-  }
-
-  def dynamicApply[This <: NonEmptyTuple, N <: Int] (self: This, n: N): Elem[This, N] = {
-    type Result = Elem[This, N]
-    val res = (self: Any) match {
-      case self: TupleXXL => self.elems(n)
-      case self: Product => self.productElement(n)
-    }
-    res.asInstanceOf[Result]
-  }
+  inline def applyStaged(n: Int): Elem[this.type, n.type] =
+    ${StagedTuple.applyStaged[this.type, n.type](
+      'this, constValueOpt[Size[this.type]], 'n, constValueOpt[n.type])}
 }
 
 @showAsInfix
 sealed class *:[+H, +T <: Tuple] extends NonEmptyTuple
 
 object *: {
-  inline def unapply[H, T <: Tuple](x: H *: T) = (x.head, x.tail)
+  inline def unapply[H, T <: Tuple](x: H *: T) =
+    (scala.runtime.DynamicTuple.dynamicHead(x), scala.runtime.DynamicTuple.dynamicTail(x))
```

Comment:
This change is only for staging, it should be
```scala
if (stageIt) (scala.runtime.DynamicTuple.dynamicHead(x), scala.runtime.DynamicTuple.dynamicTail(x))
else (x.head, x.tail)
```

Reply:
Should be the other way around, no? `if (!stageIt)`